packages/core/src/domain/configuration/configuration.ts

```diff
@@ -30,43 +30,126 @@ export const TraceContextInjection = {
+   * @default granted
+   */
   trackingConsent?: TrackingConsent | undefined
 
   // transport options
+  /**
+   * Optional proxy URL, for example: https://www.proxy.com/path.
+   * See [Proxy Your Browser RUM Data](https://docs.datadoghq.com/real_user_monitoring/guide/proxy-rum-data) for further information.
+   */
   proxy?: string | ProxyFn | undefined
+  /**
+   * The Datadog [site](https://docs.datadoghq.com/getting_started/site) parameter of your organization.
+   * @default datadoghq.com
+   */
   site?: string | undefined
 
   // tag and context options
+  /**
+   * The service name for your application. Follows the [tag syntax requirements](https://docs.datadoghq.com/getting_started/tagging/#define-tags).
+   */
   service?: string | undefined
+  /**
+   * The application’s environment, for example: prod, pre-prod, and staging. Follows the [tag syntax requirements](https://docs.datadoghq.com/getting_started/tagging/#define-tags).
+   */
   env?: string | undefined
+  /**
+   * The application’s version, for example: 1.2.3, 6c44da20, and 2020.02.13. Follows the [tag syntax requirements](https://docs.datadoghq.com/getting_started/tagging/#define-tags).
+   */
   version?: string | undefined
 
   // cookie options
   /**
+   * Whether a secure cross-site session cookie is used
+   * @default false
    * @deprecated use usePartitionedCrossSiteSessionCookie instead
    */
   useCrossSiteSessionCookie?: boolean | undefined
+  /**
+   * Use a partitioned secure cross-site session cookie. This allows the RUM Browser SDK to run when the site is loaded from another one (iframe). Implies `useSecureSessionCookie`.
+   * @default false
+   */
   usePartitionedCrossSiteSessionCookie?: boolean | undefined
+  /**
+   * Use a secure session cookie. This disables RUM events sent on insecure (non-HTTPS) connections.
+   * @default false
+   */
   useSecureSessionCookie?: boolean | undefined
+  /**
+   * Preserve the session across subdomains for the same site.
+   * @default false
+   */
   trackSessionAcrossSubdomains?: boolean | undefined
 
   // internal options
+  /**
+   * [Internal option] Enable experimental features
+   */
   enableExperimentalFeatures?: string[] | undefined
+  /**
+   * [Internal option] Configure the dual chipping to another datacenter
+   */
   replica?: ReplicaUserConfiguration | undefined
+  /**
+   * [Internal option] Set the datacenter from where the data is dual chipped
+   */
   datacenter?: string
+  /**
+   * [Internal option] Datadog internal analytics subdomain
+   */
   // TODO next major: remove this option and replace usages by proxyFn
   internalAnalyticsSubdomain?: string
-
+  /**
+   * [Internal option] The percentage of telemetry configuration sent. A value between 0 and 100.
+   * @default 5
+   */
   telemetryConfigurationSampleRate?: number
+  /**
+   * [Internal option] The percentage of telemetry usage sent. A value between 0 and 100.
+   * @default 5
+   */
   telemetryUsageSampleRate?: number
```

Comment:
**💭 thought:** ‏I'm a bit surprised to see those options are internal. I'm fine with it, I just thought those were user-facing.

Reply:
IIRC, the discussion I had with Bastien when introducing `telemetryConfigurationSampleRate`, the idea was that we didn't want to overcomplicate the telemetry on the customer side and only advocate for `telemetrySampleRate`.